R/qenv-get_code.R

```diff
@@ -11,7 +12,7 @@
 #' get_code(q3)
 #' get_code(q3, deparse = FALSE)
 #' @export
-setGeneric("get_code", function(object, deparse = TRUE) {
+setGeneric("get_code", function(object, deparse = TRUE, names = NULL) {
```

Comment:
```
> q <- new_qenv()
> q <- eval_code(q, quote(i <- iris))
> q <- eval_code(q, quote(i <- subset(i, Species != "setosa")))
> q <- eval_code(q, quote(m <- iris))
> q <- eval_code(q, quote(m <- subset(m, Species == "setosa")))
> 
> get_code(q, names = c("i", "m")) %>% cbind
     .                                      
[1,] "i <- iris"                            
[2,] "i <- subset(i, Species != \"setosa\")"
[3,] "m <- iris"                            
[4,] "m <- subset(m, Species == \"setosa\")"
```

Shouldn't code be grouped by `names`?

Reply:
All right but wouldn't it make sense in an interactive session?

Maybe another argument?

Reply:
I don't think it makes sense to have a separate argument.

Consider following code:
```
parent <- get_data("parent")
child1 <- get_data("child")
child2 <- get_data("child2")
child3 <- get_data("child3")
parent <- filter(parent, id %in% c(<some ids>)
child1 <- filter(child1, parent_id %in% parent$id)
child2 <- filter(child2, parent_id %in% parent$id)
child3 <- filter(child3, parent_id %in% parent$id)
```

`get_code(qenv, c("parent", "child1", "child2"))` should return:

```r
parent <- get_data("parent")
child1 <- get_data("child")
child2 <- get_data("child2")
parent <- filter(parent, id %in% c(<some ids>)
child1 <- filter(child1, parent_id %in% parent$id)
child2 <- filter(child2, parent_id %in% parent$id)
```


What would you expect when grouped?
```
# $parent code:
parent <- get_data("parent")
parent <- filter(parent, id %in% c(<some ids>)

# $child1 code:
parent <- get_data("parent")
child1 <- get_data("child")
parent <- filter(parent, id %in% c(<some ids>)
child1 <- filter(child1, parent_id %in% parent$id)

# $child2 code:

parent <- get_data("parent")
child2 <- get_data("child2")
parent <- filter(parent, id %in% c(<some ids>)
child2 <- filter(child2, parent_id %in% parent$id)
```
To achieve grouped code one can:

```
lapply(c("parent", "child1", "child2"), function(x) {
  get_code(qenv, x)
})
```

Reply:
I was following an assumption (based on the documentation) that a named vector is returned. Perhaps the assumption was a false one.

Reply:
sorry, updated the documentation. resolving this conversation